hips/hip-9997.md

```diff
@@ -0,0 +1,197 @@
+
+### 6. `helm chart` is removed, integrated into rest of CLI
+
+Wherever possible, the subcommands provided by the new `helm chart` command should be integrated into existing Helm CLI commands.
+
+`helm chart pull` should be baked into `helm pull`, `helm chart push` should be `helm push` (new).
+
+Commands that work with the cache will be removed: `helm chart save`, `helm chart remove` and `helm chart export`.
+
+`helm registry` will be kept as is. This manages auth against OCI registries.
+
+
+### 7. Experimental until clear messaging from OCI
+
+Considering that the rest of the items above are implemented and resolved, the OCI feature set will not be made generally available until there is clear messaging from the Open Container Initiative (OCI) regarding the way that Helm is using registries.
+
+The technical details of how Helm is using the OCI distribution spec must be validated in writing, in a document approved by OCI, clearly stating that this is the correct way to publish an arbitrary artifact. Only after this will these features be taken out of experimental mode and made generally available.
+
+## Backwards compatibility
+
+Since the existing feature set is currently experimental, there will be no promise of backwards compatibility with prior OCI support (sorry!)
+
+## Security implications
+
+Registry authentication introduces a new attack vector. Upon running `helm registry login`, these credentials are stored in an unencrypted JSON file. With that said, the underlying code is leveraging docker/cli, which has the ability to store these credentials into the system keychain instead.
+
+So, while a new attack vector has been introduced, it should not be considered any less safe than using `docker login`.
+
+Note: this is already implemented in the current OCI feature set.
+
+## How to teach this
+
+Following the implementation of all facets of this HIP, extensive documentation should be added to the Helm website on how to leverage OCI.
+
+Additionally, sites containing community charts, such as [Artifact Hub](https://artifacthub.io/), should enable providers to distribute charts over OCI, and display friendly copy-paste instructions for how to download an OCI-based chart.
+
+## Reference implementation
+
+Please see [the Helm documentation](https://helm.sh/docs/topics/registries/) for existing OCI support.
+
+## Rejected ideas
+
+N/A
+
+## Open issues
+
+The issues below are still unresolved.
+
+### What is the correct media type to use for the chart content layer?
+
+The media type for the chart content layer is currently `application/tar+gzip`, however it has been pointed out that [this is not an official type in the IANA database](https://www.iana.org/assignments/media-types/media-types.xhtml).
+
+Either a.) we go through with registering the media type through IANA, or b.) use a unique media type. One alternative could be `application/vnd.cncf.helm.chart.content.layer.v1+tar`.
```

Comment:
Considering the IANA concerns, I'd prefer option b here. Docker's image layers follow the same convention.

Reply:
👍